packages/runtime/src/widgets/Form/MultiSelect.tsx

```diff
@@ -47,21 +47,28 @@ const MultiSelect: React.FC<MultiSelectProps> = (props) => {
   const { data, ...rest } = props;
   const [options, setOptions] = useState<SelectOption[]>([]);
   const [newOption, setNewOption] = useState("");
-  const [selectedValues, setSelectedValues] = useState<string[] | undefined>(
-    isString(props.value) ? [props.value] : props.value,
-  );
+  const [selectedValues, setSelectedValues] = useState<string[]>();
 
   const action = useEnsembleAction(props.onItemSelect);
   const { rawData } = useTemplateData({ data });
   const { id, rootRef, values } = useRegisterBindings(
-    { ...rest, selectedValues, options },
+    { ...rest, initialValue: props.value, selectedValues, options },
     props.id,
     {
       setSelectedValues,
       setOptions,
     },
   );
 
+  useEffect(() => {
+    if (!selectedValues) {
+      const initVal = isString(values?.initialValue)
+        ? [values?.initialValue]
+        : values?.initialValue;
+      setSelectedValues(initVal as string[]);
+    }
```

Comment:
I think for this one `selectedValues` is the initial value.

Reply:
No, actually I did not assign the initial value in local state while state initiate, we are updating state value after initial value from the props are being evaluated by the `useRegisterBindings` only

Reply:
I meant that `initialValue` should be derived from `selectedValues` not the `value` prop.

Reply:
But on the first render, `selectedValues` will be empty array and we will assign the initialValues,  and when we change the any value of `Multiselect`, this time we only need to update the selectedValues, no need to reassign.